src/rpdk/core/project.py

```diff
@@ -293,6 +296,141 @@ def submit(
                     f, endpoint_url, region_name, role_arn, use_role, set_default
                 )
 
+    def generate_docs(self):
+        # generate the docs folder that contains documentation based on the schema
+        docs_path = "{}/docs".format(self.root)
+
+        if not self.type_info or not self.schema or "properties" not in self.schema:
+            LOG.warning(
+                "Could not generate schema docs due to missing type info or schema"
+            )
+            return
+
+        if os.path.isdir(docs_path):
+            LOG.debug("Docs directory already exists, recreating...")
+            shutil.rmtree(docs_path)
+        os.mkdir(docs_path)
+
+        LOG.debug("Writing generated docs")
+
+        docs_schema = copy.deepcopy(
+            self.schema
+        )  # take care not to overwrite master schema
```

Comment:
we've been debating this, the safer approach might be to serialize and deserialize to json. this is because some objects may not support deep copying (ref proxies), although we don't currently use them. but it's a fairly common idiom in the codebase:

```bash
$ git grep deepcopy
$ git grep 'json.loads(json.dumps('
src/rpdk/core/contract/resource_client.py:        schema = json.loads(json.dumps(self._schema))
src/rpdk/core/contract/resource_client.py:        schema = json.loads(json.dumps(self._schema))
tests/jsonutils/test_inliner.py:    copy = json.loads(json.dumps(schema))
tests/test_data_loaders.py:    copy = json.loads(json.dumps(BASIC_SCHEMA))
```

Reply:
Neat trick! Changed per your suggestion.